src/messages/variant.rs

```diff
@@ -115,6 +121,19 @@ pub(crate) enum Variant {
         nonce: [u8; 32],
         nonce_signature: Signature,
     },
+    /// Message sent by dst to indicate that Src is ahead in knowledge.
+    /// A follow-up reply will be sent by Src with SectionKnowledge
+    SrcAhead {
+        key: bls::PublicKey,
+        msg: Box<Message>,
+    },
+    /// Message sent by dst to indicate that sender is lagging on knowledge and shares it.
+    // SrcOutdated(Variant),
+    /// Message sent by dst to indicate that the Dst is ahead on knowledge and shares it.
+    DstAhead(SectionChain),
```

Comment:
It seems we neither need `DstAhead` nor `SrcAhead` ?, if a node is willing to give other its up to date info it can simply send a `SectionKnowledge`, and I presume this is done only as a response to a query to obtain up to date information?

Reply:
@Yoga07 ^^

Reply:
> I presume this is done only as a response to a query to obtain up to date information?

Aye true, that was the intention. But, it turns out we don't actually need a separate variant for this and we can simply fire `SectionKnowledge` to update the Src, the message would have reached consensus. Therefore we don't need `DstAhead`.

Though we do need SrcAhead to inform them of the lag we are facing and it acts as the trigger to send up updated information.

Reply:
> Though we do need SrcAhead to inform them of the lag we are facing and it acts as the trigger to send up updated information.

Yes, this was my impression too, for replying with an error we can use the `SectionKnowledge`, but when a node needs to send the query (after it processed a message that made it realise it was lagging) to get the new keys then we need a diff type of message which I supposed was this one.